kpatch-build/create-diff-object.c

```diff
@@ -2100,10 +2101,14 @@ static int printk_index_group_size(struct kpatch_elf *kelf, int offset)
 
 static int parainstructions_group_size(struct kpatch_elf *kelf, int offset)
 {
-	static int size = 0;
-	char *str;
+	static int size = 0, omit = 0;
+	char *str, *omit_str;
 
-	if (!size) {
+	omit_str = getenv("OMIT_PARA");
+	if (omit_str)
+		omit = atoi(omit_str);
+
+	if (!size && !omit) {
```

Comment:
Does this work out if you omitted this "omit" code and left create-diff-object untouched?  If I read the code correctly, the group_size callbacks are only executed when their sections are encountered:

```
kpatch_process_special_sections()
  	for (special = special_sections; special->name; special++) {
		...
		sec = find_section_by_name(&kelf->sections, special->name);
		if (!sec || !sec->rela)
			continue;

		kpatch_regenerate_special_section(kelf, lookup, special, sec->rela);

			group_size = special->group_size(kelf, src_offset);
```

So I'd expect create-diff-object would only care about PARA_STRUCT_SIZE when the corresponding .parainstructions section is present.  (Which I believe torvalds/linux@f7af6977 removes.)

Reply:
You are correct, my bad do not need to even pass omit